Core/src/org/sleuthkit/autopsy/report/FileReportSettings.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Autopsy Forensic Browser
+ *
+ * Copyright 2019 Basis Technology Corp.
+ * Contact: carrier <at> sleuthkit <dot> org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sleuthkit.autopsy.report;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Class for persisting the file properties used by FileReportGenerator to drive
+ * report generation by FileReportModules.
+ */
+class FileReportSettings implements Serializable  {
+    
+    private static final long serialVersionUID = 1L;
+    private Map<FileReportDataTypes, Boolean> filePropertiesInfo = new HashMap<>();
```

Comment:
I'm thinking that we can simplify this Map to a Set. The FileReportGenerator only needs to know which types are enabled, and a Set would provide an O(1) lookup as the generator iterated through the available types. I am also wondering if we will want an "all" flag like we will need for artifact types and tag types for the TableReportGenerator. A nice benefit of the Set and the "all " flag is that we can then easily accommodate new data types in future releases.

Reply:
Same as with TableReportSettings - I will copy/paste your note into the design document so that it is not lost. I think actual work on this should be done as part of the ReportGeneration refactoring story. I'm trying to make only minimal necessary code changes at this point to keep everything working "as before".